.gitmodules

```diff
@@ -0,0 +1,9 @@
+[submodule "thirdparty/fastrtps"]
+	path = thirdparty/fastrtps
+	url = https://github.com/eProsima/Fast-RTPS.git
+[submodule "thirdparty/foonathan_memory_vendor"]
+	path = thirdparty/foonathan_memory_vendor
+	url = https://github.com/eProsima/foonathan_memory_vendor.git
+[submodule "thirdparty/fastcdr"]
+	path = thirdparty/fastcdr
+	url = https://github.com/eProsima/Fast-cdr.git
```

Comment:
afaik git submodules tend to be error prone as they require some good git knowledge and discipline, specially when trying to keep versions between branches synced, any alternative to this we can think of?

Reply:
That's true, the actual use of *git submodules* is temporal. We are analysing the idea to manage these dependencies with the *superbuild* pattern of `cmake`.

Reply:
*xtypes* could be separated in a future. At this moment, we have the library inside because we are developing `fastrtps` over it, and xtypes is an integral part of it. An advantage of it is that as `ros2` use `fastrtps` inside, the `soss-ros2` system handle could avoid the conversion phase because its native message can propagate to soss directly. The same can be applied to `soss-dds` system handle.

Anyway, can be a topic of discussion for the next meeting.

Reply:
I believe if we could have separate code for `xtypes` would be the best as `soss` is not (and shouldn't be) necessarily bounded to`ros2`, so it's dependency on `fastrtps` just for `xtypes` becomes a bit of an overkill solution imho.

Reply:
*xtypes* is mainly inside of *fastrtps* because from the *fastrtps* view it is a core part and the  *fastrtps* development is easier with this module inside. Anyway, we understand perfectly the *soss* view in this regard. We are currently working in to extract *xtypes* from *fastrtps* to offer this feature without their dependencies. I hope in the next few days to offer a SOSS commit only with *xtypes* dependecy.

Reply:
That sounds great, thanks!

Reply:
The newly added commit uses *xtypes* as the only third-party. In the future, and in order to avoid git management of submodules, we can add it as a *CMake ExternalProject* and download it automatically.

Reply:
Yup, that sounds like a good idea.